Git-commit: b8bd1ee0e845c7365c018c40e3575a329d47bbfc
From: Naushir Patuck <naush@raspberrypi.com>
Date: Wed, 2 Dec 2020 16:48:41 +0000
Subject: [PATCH] media: bcm2835-unicam: Clear clock state when stopping
 streaming
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.5.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

raspberrypi inclusion
category: feature
bugzilla: 50432

--------------------------------

Commit 65e08c465020d4c5b51afb452efc2246d80fd66f failed to clear the
clock state when the device stopped streaming. Fix this, as it might
again cause the same problems when doing an unprepare.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
Signed-off-by: Fang Yafen <yafen@iscas.ac.cn>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 drivers/media/platform/bcm2835/bcm2835-unicam.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/media/platform/bcm2835/bcm2835-unicam.c b/drivers/media/platform/bcm2835/bcm2835-unicam.c
index 60309de02369..a8b3f5433f04 100644
--- a/drivers/media/platform/bcm2835/bcm2835-unicam.c
+++ b/drivers/media/platform/bcm2835/bcm2835-unicam.c
@@ -1769,6 +1769,7 @@ static void unicam_stop_streaming(struct vb2_queue *vq)
 
 			clk_disable_unprepare(dev->vpu_clock);
 			clk_disable_unprepare(dev->clock);
+			dev->clocks_enabled = false;
 		}
 		unicam_runtime_put(dev);
 
-- 
2.31.1

